### ISSUE TITLE
support for gamma deviance

### ISSUE THREAD

**Issue by spedygiorgio** [catboost/catboost#2795]
Problem:
catboost version: 1.2.7
Operating System: Windows 11
CPU: intel i9
GPU: nvidia rxt a200

I would suggest the implementation of gamma loss for regression problem, see e.e. [sklearn gamma deviance](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_gamma_deviance.html)
